### PR TITLE
Feature/13 button

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
**#13 Buttons**
Issues Covered- **#14 , #15 , #16 , #17** 
-Create Button 
-Add Property to enable Floating Button.
-Add Different states of on click , disabled state.
-Add Properties for Commands , modifying the color of the elements, modifying  radius .
-Add different types of buttons to UI of the Demo APP.

### CODE REVIEW

File: Trimble.Modus.Components/Controls/Button/TMButton.cs [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+
+                        break;
+                    case ButtonSize.Small:
+                        tmButton._titleLabel.FontSize = 14;
+                        if (tmButton.imageSet)
+                        {
+                            tmButton._titleLabel.Padding = new Thickness(0, 8, 16, 8);
+                            tmButton._iconImage.Margin = new Thickness(12, 8, 8, 8);
+                            tmButton._iconImage.IsVisible = true;
+
```
Comment (BAL4-R) [inline Trimble.Modus.Components/Controls/Button/TMButton.cs:313]: Extra empty lines can be removed

File: Trimble.Modus.Components/Controls/Button/TMButton.cs [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+                            tmButton._iconImage.Margin = new Thickness(16, 16, 8, 16);
+                            tmButton._iconImage.IsVisible = true;
+                        }
+                        else
+                        {
+                            tmButton._titleLabel.Padding = new Thickness(24, 16, 24, 16);
+                            tmButton._iconImage.IsVisible = false;
+
+                        }
+
```
Comment (BAL4-R) [inline Trimble.Modus.Components/Controls/Button/TMButton.cs:342]: Empty line can be removed

File: Trimble.Modus.Components/Controls/Button/TMButton.cs [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+            }
+        }
+
+        private static void OnIsDisabledChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var button = (TMButton)bindable;
+
+            if ((bool)newValue)
+            {
+                button.frame.BackgroundColor = Color.FromArgb("CEDEEB");
```
Comment (BAL4-R) [inline Trimble.Modus.Components/Controls/Button/TMButton.cs:358]: Can the color be added to dictionary?

File: Trimble.Modus.Components/Controls/Button/TMButton.cs [hunk 4]
```diff
@@ -0,0 +1,407 @@
+﻿
+using Microsoft.Maui.Controls.Shapes;
+
+
+namespace Trimble.Modus.Components
+{
+
+    public class TMButton : ContentView
```
Comment (BAL4-R) [inline Trimble.Modus.Components/Controls/Button/TMButton.cs:8]: Can you add "Clicked" property for button

File: Trimble.Modus.Components/Controls/Button/TMButton.cs [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+        public new static readonly BindableProperty BackgroundColorProperty = 
+            BindableProperty.Create(nameof(BackgroundColor), typeof(Color), typeof(TMButton), (Color)BaseComponent.colorsDictionary()["TrimbleBlue"],propertyChanged:onBackgroundColorChanged);
+
+        public static readonly BindableProperty BorderColorProperty =
+            BindableProperty.Create(nameof(BorderColor), typeof(Color), typeof(TMButton), (Color)BaseComponent.colorsDictionary()["TrimbleBlue"], propertyChanged: onBorderColorChanged);
+       
+        public static readonly BindableProperty TextColorProperty = 
+            BindableProperty.Create(nameof(TextColor), typeof(Color), typeof(TMButton), defaultValue: Colors.White, propertyChanged: onTextColorChanged);
+       
+        public static readonly BindableProperty ButtonRadiusProperty =
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/Button/TMButton.cs:48]: @ElishaSamPeterPrabhu Please have this default button radius as a constant in the top of the class

File: Trimble.Modus.Components/Controls/Button/TMButton.cs [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+        {
+
+            _titleLabel = new Label
+            {
+                TextColor = Colors.White,
+            };
+
+            _iconImage = new Image
+            {
+                WidthRequest = 24,
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/Button/TMButton.cs:129]: @ElishaSamPeterPrabhu  This should be defined as a constant

File: Trimble.Modus.Components/Controls/Button/TMButton.cs [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+                }
+            };
+
+
+
+            Content = frame;
+            
+
+            GestureRecognizers.Add(_tapGestureRecognizer = new TapGestureRecognizer());
+            _tapGestureRecognizer.Tapped += OnTapped;
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/Button/TMButton.cs:163]: @ElishaSamPeterPrabhu  We should unsubscribe in the Dispose method.

File: Trimble.Modus.Components/Controls/Button/TMButton.cs [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+                }
+            }
+        }
+
+        private static void OnSizeChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            if (bindable is TMButton tmButton && !tmButton.IsFloatingButton)
+            {
+                var size = (ButtonSize)newValue;
+                Console.WriteLine($"Size OnSize: {size}");
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/Button/TMButton.cs:285]: Please remove the log

File: Trimble.Modus.Core/BaseComponent.cs [hunk 9]
```diff
@@ -1,8 +1,12 @@
-﻿namespace Trimble.Modus.Core
```
Comment (rthanga1) [inline Trimble.Modus.Core/BaseComponent.cs:1]: Please remove the .Core project if it is not necessary

File: Trimble.Modus.Components/Controls/Button/TMButton.cs [hunk 10]
```diff
@@ -0,0 +1,407 @@
...
+            var button = (TMButton)bindable;
+
+            if ((bool)newValue)
+            {
+                button.frame.BackgroundColor = Color.FromArgb("CEDEEB");
+
+                button._iconImage.Opacity = 0.5;
+                button.GestureRecognizers.Clear();
+
+                button.GestureRecognizers.Clear();
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/Button/TMButton.cs:363]: Same line is repeated

File: Trimble.Modus.Components/Controls/Button/TMButton.cs (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/TMButton.cs]: Removed Empty line

File: Trimble.Modus.Components/Controls/Button/TMButton.cs (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/TMButton.cs]: Removed Empty line

File: Trimble.Modus.Components/Controls/Button/TMButton.cs (hunk 10 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/TMButton.cs]: Removed Repeated line

File: Trimble.Modus.Core/BaseComponent.cs (hunk 9 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Core/BaseComponent.cs:1]: Removed .Core Project

File: Trimble.Modus.Components/Controls/Button/TMButton.cs (hunk 6 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/TMButton.cs]: Defined as Constant

File: Trimble.Modus.Components/Controls/Button/TMButton.cs (hunk 5 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/TMButton.cs:51]: Defined as Constant

File: Trimble.Modus.Components/Controls/Button/TMButton.cs (hunk 4 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/TMButton.cs:8]: Added Clicked Property

File: Trimble.Modus.Components/Controls/Button/TMButton.cs (hunk 7 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/TMButton.cs]: OnDispose Method not found some other way must be implemented to unsubscribe

**Review by rthanga1 — APPROVED** [review]

**Review by BAL4-R — APPROVED** [review]

**Pull request merged**